state/protocol/kvstore.go

```diff
@@ -113,8 +216,64 @@ type VersionedEncodable interface {
+	ActivationView uint64
+}
+
+// UpdatableField represents a protocol parameter which can be updated using a ViewBasedActivator.
+type UpdatableField[T any] struct {
+	// CurrentValue is the value that is active after constructing the block
+	// that this Protocol State pertains to.
+	CurrentValue T
+
+	// Update is optional and is nil if no value update has been scheduled yet.
+	// This field will hold the last scheduled update until a newer update
+	// directive is received, even if the value update has already happened.
+	// The update should be applied when reaching or exceeding the ActivationView.
+	Update *ViewBasedActivator[T]
+}
+
+// MagnitudeOfChangeVersion is intended as an intuitive representation of the “magnitude of change”.
+//
+// # CAUTION: Don't confuse this with semver!
+//
+// This versioning representation DEVIATES from established Semantic Versioning.
+// Any two different versions of the Execution Stack are considered incompatible.
+// In particular, two versions only differing in their minor, might be entirely downwards-INCOMPATIBLE.
+//
+// We generally recommend to use Integer Versioning for components. The MagnitudeOfChangeVersion scheme should
+// be only used when there is a clear advantage Integer Versioning which outweighs the risk of falsely
+// making compatibility assumptions by confusing this scheme with Semantic Versioning!
+//
+// MagnitudeOfChangeVersion helps with an intuitive representation of the “magnitude of change”.
+// For example, for the execution stack, bug fixes closing unexploited edge-cases will be a relatively
+// frequent cause of upgrades. Those bug fixes could be reflected by minor version bumps, whose
+// imperfect downwards compatibility might frequently suffice to warrant Access Nodes using the same
+// version (higher minor) across version boundaries. In comparison, major version change would generally
+// indicate broader non-compatibility (or larger feature additions) where it is very unlikely that the Access
+// Node can use one implementation for versions with different major.
+//
+// We emphasize again that this differentiation of “imperfect but good-enough downwards compatibility”
+// is in no way reflected by the versioning scheme. Any automated decisions of compatibility for different
+// versions are to be avoided (including versions where only the minor is different).
+//
+// Engineering teams using this scheme must be aware that the MagnitudeOfChangeVersion is easily
+// misleading wrt to incorrect assumptions about downwards compatibility. Avoiding problems (up to and
+// including the possibility of mainnet outages) requires continued awareness of all engineers in the
+// teams working with this version. The engineers in those teams must commit to diligently documenting
+// all relevant changes, details regarding magnitude of changes and if applicable “imperfect but
+// good-enough downwards compatibility”.
+type MagnitudeOfChangeVersion struct {
```

Comment:
What are your thoughts using `MagnitudeVersion` instead of `MagnitudeOfChangeVersion` or maybe something like `FeatureVersion`?

Reply:
Done in https://github.com/onflow/flow-go/pull/7020/commits/1e3452e05059645fc04b965cc4fb1c2ac50ff649